src/utils/ProfileProvider.tsx

```diff
@@ -63,22 +70,36 @@ export default function ProfileProvider({ children }: { children: ReactNode }) {
   const [profileRoles, setProfileRoles] = useState<ProfileRole[]>([]);
   const { userId } = auth;
 
-  const loadProfile = useCallback(async () => {
-    setProfileReady(false);
+  const loadProfile = useCallback(
+    async (id?: string) => {
```

Comment:
Why did we decide to change loadProfile to take in id, and why didn't we have it before?

Reply:
we didn't have this before since the assumption was that user id will always be correct

turns out, for a synchronous execution (i.e. await login, then immediately load profile), the state change happens after load profile, resulting in `loadProfile` detecting user id to be undefined, which gets treated as not logged in, even if log in was successful

therefore, this somewhat hacky solution is simply to allow us to get the userid directly from the return of sign in, which we can then pass onto `loadProfile`.